packages/flutter/lib/src/material/tabs.dart

```diff
@@ -562,8 +562,12 @@ class TabBarSelectionState<T> extends State<TabBarSelection<T>> with SingleTicke
   @override
   void didUpdateConfig(TabBarSelection<T> oldConfig) {
     super.didUpdateConfig(oldConfig);
+    _previousValue = _value;
+    _value = config.value ?? values.first;
```

Comment:
I don't understand which is the authoritative value.  It seems like there's a setter for the value on the State and you can overwrite the value from the widget.  It seems like there are two uncoordinated writers.

Maybe the widget should have an initialValue or maybe we should remove the setter from the state?